src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -247,7 +247,7 @@ public static void AddDocumentManagementServices(this IServiceCollection service
             var errorReporter = services.GetRequiredService<IErrorReporter>();
             var projectEngineFactoryProvider = new LspProjectEngineFactoryProvider(optionsManager);
```

Comment:
Can we set up `LspProjectEngineFactoryProvider` as a singleton, and then remove this lambda and just use the simpler `AddSingleton` overload?